### PR TITLE
[feat] Report progress for query apis and sdk queries

### PR DESCRIPTION
**Pull request by mihran113**
To do:

- [x] Add progress reporting for search APIs and  SDK queries (tqdm)
- [x] Add reporting in between if there were no matches for an interval (for ex. 0.25s)
- [x] Add query param for APIs to turn off progress reporting if needed
- [x] Fix unit tests
- [x] Add tqdm progress bars for SDK queries

### CODE REVIEW

File: aim/sdk/repo.py [hunk 1]
```diff
@@ -507,17 +522,19 @@ def move_runs(self, run_hashes: List[str], dest_repo: 'Repo') -> Tuple[bool, Lis
...
              query (str): query expression.
+             report_mode(:obj:`str`, optional): indicates report mode (0: disabled, 1: tqdm progress bar, 2: tuple).
+                                                1 if not specified.
         Returns:
             :obj:`MetricCollection`: Iterable for metrics matching query expression.
         """
         self._prepare_runs_cache()
         from aim.sdk.sequences.metric import Metric
-        return QuerySequenceCollection(repo=self, seq_cls=Metric, query=query)
+        return QuerySequenceCollection(repo=self, seq_cls=Metric, query=query, report_mode=report_mode)
```
Comment (alberttorosyan) [inline aim/sdk/repo.py:537]: what about `query_images` etc?

Comment (mihran113) [reply aim/sdk/repo.py:537]: Oops, forgot about those, cause in APIs those are accessed directly and not via `repo` methods. Will update.

File: aim/sdk/sequence_collection.py [hunk 2]
```diff
@@ -139,22 +140,45 @@ def __init__(
         self,
         repo: 'Repo',
         seq_cls=Sequence,
-        query: str = ''
+        query: str = '',
+        report_mode: int = 1,
```
Comment (mahnerak) [inline aim/sdk/sequence_collection.py:144]: Define constants and replace numbers with them please.

Comment (mihran113) [reply aim/sdk/sequence_collection.py]: done ✅

**Comment by mihran113** [comment]
Resolves #1746

File: aim/web/api/runs/object_api_utils.py [hunk 3]
```diff
@@ -125,11 +125,29 @@ def _pack_run_data(run_: 'Run', traces_: list):
...
 
+        last_reported_progress_time = time.time()
+        run_info = None
+        progress_reports_sent = 0
         for run_info in self.trace_cache.values():
-            traces_list = []
-            for trace in run_info['traces']:
-                traces_list.append(self._get_trace_info(trace, True, True))
-            yield _pack_run_data(run_info['run'], traces_list)
+            if report_progress and time.time() - last_reported_progress_time:
```
Comment (alberttorosyan) [inline aim/web/api/runs/object_api_utils.py:132]: `> AIM_PROGRESS_REPORT_INTERVAL:`?

**Review by alberttorosyan — COMMENTED** [review]
left one minor comment

File: aim/web/api/runs/object_api_utils.py (hunk 3 above)
Comment (mihran113) [reply aim/web/api/runs/object_api_utils.py]: thanks for the notice, fixed

**Review by alberttorosyan — APPROVED** [review]

**Pull request merged**